app/dao/notifications_dao.py

```diff
@@ -89,11 +90,15 @@ def dao_create_notification(notification):
     db.session.add(notification)
 
 
-def _decide_permanent_temporary_failure(current_status, status):
+def _decide_permanent_temporary_failure(current_status, status, code=None):
     # Firetext will send pending, then send either succes or fail.
-    # If we go from pending to delivered we need to set failure type as temporary-failure
+    # If we go from pending to failure we need to set failure type as temporary-failure
+    # if we get a detailed code from firetext, we should use that code to set status instead
```

Comment:
I suggest capatilising this first word and adding a full stop the line above so it's clear they are two separate sentences

Reply:
They actually are not totally separate sentences. I made that clearer now.